### PR TITLE
Stable AstroNvim v2.0 Release

### PR DESCRIPTION
**Pull request by mehalter**
With the official stable release of [Neovim v0.8](https://github.com/neovim/neovim/releases/tag/v0.8.0), we are excited to announce that AstroNvim v2.0 is fully supporting of the new update!

**Please Note:** AstroNvim v2.0 will not have support for Neovim <0.8. If you are still using Neovim 0.7 and cannot update, you can set the following settings for `updater` to restrict updates from updating to v2.0:

```lua
updater = {
  channel = "stable",
  version = "v1.10.0",
}
```

# New Features

- Use new Neovim 0.8 APIs
- Hide the command line when not in use
- Use `winbar` to display LSP breadcrumbs in the active window and filename in inactive windows
- Migrate from Feline to [Heirline](https://github.com/rebelot/heirline.nvim)
- Remove `mason-tool-installer` plugin in favor of just `mason-lspconfig` and `mason-null-ls`. We found this plugin to be not very performant and not as much functionality. If you use this plugin, we have included a snippet in the migration guide for how to add it to your `user/init.lua` file.
- Add clickable elements to statusline:
  - the git branch will open the branches and let you switch branches of the current repo
  - the language server names will open `:LspInfo`
  - the git differences will open a list of changes in Telescope
  - the diagnostics will open a list of diagnostics in Telescope
  - the symbols in the LSP breadcrumbs will navigate you to the symbol definition
- Added source selector to `neo-tree` and the tabs are clickable to switch sources
- Brand new `astronvim.status` module for configuring custom statuslines (New docs are here: https://astronvim.github.io/Recipes/status)
- Deprecate old user configuration options and simplifying some things (See migration guide)
- Unify `on_attach` for `null-ls` and language servers, no need for user definition of it
- Auto format on save by default
- Remove terminal escape bindings
- Changed some bindings (See migration guide)
- Added new `UI` menu for toggling quick settings `<leader>+u`
- Add new `lsp.formatting` user options for disabling formatting, adjusting timeout, etc. (New docs addition: https://astronvim.github.io/Recipes/advanced_lsp#controlling-formatting)
- Add `highlights.init` table to add custom highlight groups that get applied to all colorschemes

# BREAKING CHANGES

This update does come with breaking changes. We have put together an extensive migration guide to the documentation to move user configurations from v1.0 to v2.0. Please read it in full: https://astronvim.github.io/Configuration/v2_migration

### CODE REVIEW

**Review by kabinspace — APPROVED** [review]

**Comment by jayp0521** [comment]
let me know if you face any bugs in `mason-null-ls`.

**Comment by mehalter** [comment]
Definitely @jayp0521 ! I was also the one a few days ago who opened the PR for adding the missing `prettier` sources to `mason-null-ls`. Thanks for your heard work putting together that plugin! It's super great!

**Comment by jayp0521** [comment]
> Definitely @jayp0521 ! I was also the one a few days ago who opened the PR for adding the missing `prettier` sources to `mason-null-ls`. Thanks for your heard work putting together that plugin! It's super great!

yes, I noticed! Thanks

File: lua/core/utils/lsp.lua [hunk 1]
```diff
@@ -0,0 +1,226 @@
+--- ### AstroNvim LSP
+--
+-- This module is automatically loaded by AstroNvim on during it's initialization into global variable `astronvim.lsp`
```
Comment (queue-tip) [inline lua/core/utils/lsp.lua:3]:
```suggestion
-- This module is automatically loaded by AstroNvim on during its initialization into global variable `astronvim.lsp`
```

**Comment by cfbender** [comment]
The `user_example` still contains config for `mason-tool-installer`

**Pull request merged**